api/src/test/java/org/openmrs/api/handler/PatientDataUnvoidHandlerTest.java

```diff
@@ -144,4 +149,36 @@ public void handle_shouldNotUnvoidTheOrdersAndEncountersThatNeverGotVoidedWithTh
 		Assert.assertNotNull(testOrder.getVoidReason());
 		
 	}
+	
+	/**
+	 * @verifies unvoid the members associated with the patient
+	 * @see PatientDataUnvoidHandler#handle(Patient,User,Date,String)
+	 */
+	@Test
+	public void handle_shouldUnvoidMembersAssociatedWithThePatient() throws Exception {
+		executeDataSet(COHORT_XML);
+		CohortService cs = Context.getCohortService();
+		Cohort cohort = cs.getCohort(2);
+		Patient patient = Context.getPatientService().getPatient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		cs.addMembershipToCohort(cohort, membership);
+		Assert.assertTrue(cohort.contains(patient));
+		
+		patient = Context.getPatientService().voidPatient(patient, "Void Reason");
+		Assert.assertTrue(patient.getVoided());
+		
+		Assert.assertTrue(membership.getVoided());
+		Assert.assertNotNull(membership.getDateVoided());
+		Assert.assertNotNull(membership.getVoidedBy());
+		Assert.assertNotNull(membership.getVoidReason());
+		
+		patient.setVoided(false);
```

Comment:
Adding order to both PatientDataUnvoidHandler and BaseUnvoidHandler works, but now there is an issue where the patient is already unvoided and I have these conditions in the CohortServiceImpl.patientUnvoided:

```
			List<CohortMembership> membersToUnvoid = cohort.getMemberships().stream()
					.filter(m -> m.getVoided() &&
							m.getPatient().getPatientId().equals(patient.getPatientId()) &&
							m.getVoidedBy().equals(voidedBy) &&
							DateUtils.isSameDay(m.getDateVoided(), dateVoided) &&
							m.getVoidReason().equals(voidReason))
```

I can only compare the patient id right now.